From 54d8fe7ae40902d6d38e670f4024092f53c14e1f Mon Sep 17 00:00:00 2001
From: Samuel Martin <s.martin49@gmail.com>
Date: Sat, 8 Mar 2014 13:19:14 +0100
Subject: [PATCH] Only blacklist ARM gcc 4.8.0 and 4.8.1

Since many ARM toolchain providers include the bug fix for PR58854 in
their latest releases based on gcc-4.8.2, then only blacklist gcc 4.8.0
and 4.8.1.

Signed-off-by: Samuel Martin <s.martin49@gmail.com>
---
 urcu/compiler.h | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/urcu/compiler.h b/urcu/compiler.h
index 1e30903..56115f1 100644
--- a/urcu/compiler.h
+++ b/urcu/compiler.h
@@ -118,10 +118,17 @@
 				+ __GNUC_PATCHLEVEL__)
 
 /*
+ * Official gcc releases from 4.8.0 to 4.8.2 have the following bug,
+ * however, many arm toolchain providers have the included the fix for
+ * their latest 4.8.2 releases.
+ * So, we only blacklist gcc 4.8.0 and 4.8.1.
+ * Unfortunately, this bug is not easy to test, so we rely on the
+ * knowledge of the user on its compiler.
+ *
  * http://gcc.gnu.org/bugzilla/show_bug.cgi?id=58854
  */
 # ifdef __ARMEL__
-#  if URCU_GCC_VERSION >= 40800 && URCU_GCC_VERSION <= 40802
+#  if URCU_GCC_VERSION >= 40800 && URCU_GCC_VERSION <= 40801
 #   error Your gcc version produces clobbered frame accesses
 #  endif
 # endif
-- 
1.9.0

